libs/zstbi/src/zstbi.zig

```diff
@@ -557,10 +558,57 @@ extern fn stbi_write_jpg_to_func(
     quality: c_int,
 ) c_int;
 
-test "zstbi.basic" {
-    init(std.testing.allocator);
+test "zstbi basic" {
+    init(testing.allocator);
     defer deinit();
 
-    var image = try Image.createEmpty(64, 64, 4, .{});
-    defer image.deinit();
+    var im1 = try Image.createEmpty(8, 6, 4, .{});
+    defer im1.deinit();
+
+    try testing.expect(im1.width == 8);
+    try testing.expect(im1.height == 6);
+    try testing.expect(im1.num_components == 4);
+}
+
+test "zstbi resize" {
+    init(testing.allocator);
+    defer deinit();
+
+    var im1 = try Image.createEmpty(32, 32, 4, .{});
+    defer im1.deinit();
+
+    var im2 = try Image.createEmpty(8, 6, 3, .{});
+    defer im2.deinit();
+
+    try testing.expect(im2.width == 8);
+    try testing.expect(im2.height == 6);
+    try testing.expect(im2.num_components == 3);
+}
+
+test "zstbi write and load file" {
+    init(testing.allocator);
+    defer deinit();
+
+    var img = try Image.createEmpty(8, 6, 4, .{});
+    defer img.deinit();
+
+    try img.writeToFile("test_img.png", ImageWriteFormat.png);
+    try img.writeToFile("test_img.jpg", .{ .jpg = .{ .quality = 80 } });
+
+    var img_png = try Image.loadFromFile("test_img.png", 0);
+    defer img_png.deinit();
+
+    try testing.expect(img_png.width == img.width);
+    try testing.expect(img_png.height == img.height);
+    try testing.expect(img_png.num_components == img.num_components);
+
+    var img_jpg = try Image.loadFromFile("test_img.jpg", 0);
+    defer img_jpg.deinit();
+
+    try testing.expect(img_jpg.width == img.width);
+    try testing.expect(img_jpg.height == img.height);
+    try testing.expect(img_jpg.num_components == 3); // RGB JPEG
+
+    try std.fs.cwd().deleteFile("test_img.png");
+    try std.fs.cwd().deleteFile("test_img.jpg");
```

Comment:
`writeToFn` with a fn that fills a buffer and `loadFromMemory` that reads it would be better here so the tests don't have to touch the filesystem

Reply:
I would still test the write to disk somewhat. If you want, I can comment all that test, even if it passes. Or we can test only JPG, or only PNG, so we don't touch the disk as much?

I honestly don't know how to call the `writeToFn` function at all... Where does the `ctx: ?*anyopaque` come from in Zig? I mean, I can see it in C.

Reply:
Ok, let's keep the the write to file.

Re `writeToFn`, `ctx` is just a "user data pointer", you pass a type erased pointer to anything you like or null and it will be passed back to you in your callback. This provides a means to access anything you might need in your `write_fn`, for example (untested/pseudocode):

```zig
    ...
    
    try writeToFn(
        image,
        &writeImage,
        @alignCast(@ptrCast(&writer)),
        .{ .jpg = .{ .quality = 80 },
    );
}

export fn writeImage(ctx: ?*anyopaque, data: ?*anyopaque, size: c_int) callconv(.C) void {
    const writer: *std.io.Writer = @alignCast(@ptrCast(ctx.?));
    writer.writeAll(data.?[0..@intCast(size)]) catch |err| {
        std.log.err("Failed to write image with error: {s}", .{@errorName(err)});
    };
}
```

Reply:
Perfect! All done :)

Reply:
Thanks!